packages/web3-eth/src/validation.ts

```diff
@@ -103,3 +119,121 @@ export function isTransactionCall(value: TransactionCall): boolean {
+		if (
+			transaction.chainId !== undefined &&
+			transaction.chainId !== transaction.common.customChain.chainId
+		)
+			throw new ChainIdMismatchError({
+				txChainId: transaction.chainId,
+				customChainId: transaction.common.customChain.chainId,
+			});
+	}
+
+};
+
+export const validateChainInfo = (transaction: Transaction) => {
+	if (
+		transaction.common !== undefined &&
+		transaction.chain !== undefined &&
+		transaction.hardfork !== undefined
+	) {
+		throw new CommonOrChainAndHardforkError();
+	}
+	if (
+		(transaction.chain !== undefined && transaction.hardfork === undefined) ||
+		(transaction.hardfork !== undefined && transaction.chain === undefined)
+	)
+		throw new MissingChainOrHardforkError({
+			chain: transaction.chain,
+			hardfork: transaction.hardfork,
+		});
+};
+
+export const validateLegacyGas = (transaction: Transaction<HexString>) => {
+	if (
+		// This check is verifying gas and gasPrice aren't less than 0.
+		// transaction's number properties have been converted to HexStrings.
+		// JavaScript doesn't handle negative hex strings e.g. -0x1, but our
+		// numberToHex method does. -0x1 < 0 would result in false, so we must check if
+		// hex string is negative via the inclusion of -
+		transaction.gas === undefined ||
+		transaction.gasPrice === undefined ||
+		transaction.gas.startsWith('-') ||
+		transaction.gasPrice.startsWith('-')
+	)
+		throw new InvalidGasOrGasPrice({
+			gas: transaction.gas,
+			gasPrice: transaction.gasPrice,
+		});
+	if (transaction.maxFeePerGas !== undefined || transaction.maxPriorityFeePerGas !== undefined)
+		throw new UnsupportedFeeMarketError({
+			maxFeePerGas: transaction.maxFeePerGas,
+			maxPriorityFeePerGas: transaction.maxPriorityFeePerGas,
+		});
+};
+
+export const validateFeeMarketGas = (transaction: Transaction<HexString>) => {
+	// These errors come from 1.x, so they must be checked before
+	// InvalidMaxPriorityFeePerGasOrMaxFeePerGas to throw the same error
+	// for the same code executing in 1.x
+	if (transaction.gasPrice !== undefined && transaction.type === '0x2')
+		throw new Eip1559GasPriceError(transaction.gasPrice);
+	if (transaction.type === '0x0' || transaction.type === '0x1')
+		throw new UnsupportedFeeMarketError({
+			maxFeePerGas: transaction.maxFeePerGas,
+			maxPriorityFeePerGas: transaction.maxPriorityFeePerGas,
+		});
+
+	if (
+		transaction.maxFeePerGas === undefined ||
+		transaction.maxPriorityFeePerGas === undefined ||
+		transaction.maxFeePerGas.startsWith('-') ||
+		transaction.maxPriorityFeePerGas.startsWith('-')
+	)
+		throw new InvalidMaxPriorityFeePerGasOrMaxFeePerGas({
+			maxPriorityFeePerGas: transaction.maxPriorityFeePerGas,
+			maxFeePerGas: transaction.maxFeePerGas,
+		});
+};
+
+/**
+ * This method checks if all required gas properties are present for either
+ * legacy gas (type 0x0 and 0x1) OR fee market transactions (0x2)
+ */
+export const validateGas = (transaction: Transaction<HexString>) => {
```

Comment:
In all these validator functions, where ever possible, use `web3-validator` instead of structural conditions.